connectors/mongo/integrity.go

```diff
@@ -13,19 +13,48 @@ import (
 	"log/slog"
 
 	"github.com/adiom-data/dsync/protocol/iface"
+	"github.com/cespare/xxhash"
 )
 
-func (mc *BaseMongoConnector) IntegrityCheck(ctx context.Context, flowId iface.FlowID, task iface.ReadPlanTask) (iface.ConnectorDataIntegrityCheckResult, error) {
-	filter := createFindFilter(task)
-	collection := mc.Client.Database(task.Def.Db).Collection(task.Def.Col)
-	c, err := collection.CountDocuments(mc.Ctx, filter)
+func (mc *BaseMongoConnector) IntegrityCheck(ctx context.Context, query iface.IntegrityCheckQuery) (iface.ConnectorDataIntegrityCheckResult, error) {
+	filter := createFindFilter(query)
+	collection := mc.Client.Database(query.Db).Collection(query.Col)
+
+	if query.CountOnly {
+		c, err := collection.CountDocuments(mc.Ctx, filter)
```

Comment:
Need to use the estimated count instead:https://www.mongodb.com/docs/manual/reference/method/db.collection.estimatedDocumentCount/

CountDocuments() does a full index scan